javalib/src/main/scala/java/util/AbstractSet.scala

```diff
@@ -0,0 +1,46 @@
+package java.util
+
+import scala.annotation.tailrec
+
+abstract class AbstractSet[E]() extends java.util.AbstractCollection[E] with java.util.Set[E] {
+	override def equals(o: Any) = {
+		if (o==this) {
```

Comment:
On the logic side: `o == this` will basically call `o.equals(this)`. This is likely to go into an infinite recursion.